src/Compilers/.editorconfig

```diff
@@ -12,6 +12,15 @@ dotnet_diagnostic.RS0101.severity = none
 # RS0102: Braces must not have blank lines between them
 dotnet_diagnostic.RS0102.severity = none
 
+# IDE0059: Unnecessary assignment to a value
+dotnet_diagnostic.IDE0059.severity = none
+
+dotnet_diagnostic.IDE2000.severity = none
+dotnet_diagnostic.IDE2001.severity = none
+dotnet_diagnostic.IDE2002.severity = none
+dotnet_diagnostic.IDE2003.severity = none
+dotnet_diagnostic.IDE2004.severity = none
```

Comment:
compiler was disabling the RS.... diagnostics above.  It seems liek a pity, but i think we have to do the same with the IDE2XXX diagnostics until we can decide if we want to be consistent here across the codebase or not.